crates/tags/src/storage/memory.rs

```diff
@@ -29,18 +29,25 @@ impl Storage for Memory {
     type Error = StatusCode;
 
     async fn tags(&self) -> Result<Vec<Tag>, Self::Error> {
-        todo!()
+        let our_map = self.0.read().await;
+        Ok(our_map.keys().cloned().collect())
     }
 
     async fn del(&self, tag: Tag) -> Result<(), Self::Error> {
-        todo!()
+        let mut our_map = self.0.write().await;
+        our_map.remove(&tag).ok_or(StatusCode::NotFound)?;
+        Ok(())
```

Comment:
Not sure we need delete, but we probably want yanking @npmccallum ?

Reply:
I figured I'd remove it from the http server side, but leave it in `Storage` in case some other mechanism wants/needs to delete it later.

Reply:
@rvolosatovs There are two forms of removal:

1. yanking - semver tags are excluded from results in the semver query API (TBD).
2. revocation - deletes a tree, but leaves any tags still pointing to it.

Tags are always immutable. Once they are created, they exist forever.